File: lib/domain/usecase/import_book_use_case.dart
import '../../core/model/book_metadata.dart';
import '../repository/book_repository.dart';

class ImportBookUseCase {
  final BookRepository _bookRepository;

  ImportBookUseCase(this._bookRepository);

  Future<BookMetadata> call(String sourceUri) =>
      _bookRepository.importBook(sourceUri);
}

File: domain/src/main/kotlin/com/readler/domain/usecase/ImportBookUseCase.kt
package com.readler.domain.usecase

import com.readler.core.model.BookMetadata
import com.readler.domain.repository.BookRepository

class ImportBookUseCase(
    private val bookRepository: BookRepository
) {
    suspend operator fun invoke(sourceUri: String): BookMetadata = bookRepository.importBook(sourceUri)
}
